src/Analysis/Ast/Impl/Specializations/Typing/Types/GenericTypeParameter.cs

```diff
@@ -33,17 +35,35 @@ public GenericTypeParameter(string name, IPythonModule declaringModule, IReadOnl
         public override PythonMemberType MemberType => PythonMemberType.Generic;
         public override bool IsSpecialized => true;
 
-
         public static IPythonType FromTypeVar(IArgumentSet argSet, IPythonModule declaringModule, IndexSpan location = default) {
             var args = argSet.Values<IMember>();
+            var eval = argSet.Eval;
+            var callExpression = argSet.CallExpression; 
+            var callLocation = callExpression?.GetLocation(eval?.Module);
+
             if (args.Count == 0) {
```

Comment:
Since method is getting long, how's about making a separate method like `CheckTypeVarArguments(...)`

Reply:
Done!